platforms/software/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -184,20 +183,26 @@ private AuthScheme getAuthScheme(final Collection<Authentication> authentication
 
     private void configureProxy(HttpClientBuilder builder, CredentialsProvider credentialsProvider, HttpSettings httpSettings) {
         HttpProxySettings.HttpProxy httpProxy = httpSettings.getProxySettings().getProxy();
-        HttpProxySettings.HttpProxy httpsProxy = httpSettings.getSecureProxySettings().getProxy();
+        if (httpProxy != null && httpProxy.credentials != null) {
+            AllSchemesAuthentication authentication = getAuthForProxy(httpProxy);
+            useCredentials(credentialsProvider, Collections.singleton(authentication));
+        }
 
-        for (HttpProxySettings.HttpProxy proxy : Lists.newArrayList(httpProxy, httpsProxy)) {
-            if (proxy != null) {
-                if (proxy.credentials != null) {
-                    AllSchemesAuthentication authentication = new AllSchemesAuthentication(proxy.credentials);
-                    authentication.addHost(proxy.host, proxy.port);
-                    useCredentials(credentialsProvider, Collections.singleton(authentication));
-                }
-            }
+        HttpProxySettings.HttpProxy httpsProxy = httpSettings.getSecureProxySettings().getProxy();
+        if (httpsProxy != null && httpsProxy.credentials != null) {
+            AllSchemesAuthentication authentication = getAuthForProxy(httpsProxy);
+            useCredentials(credentialsProvider, Collections.singleton(authentication));
```

Comment:
💭 Can all of this be moved into a new `useCredsForProxy` method, removing the need for `getAuthForProxy` as a separate method?

Reply:
Done